src/segment_dotnet.go

```diff
@@ -30,7 +30,7 @@ func (d *dotnet) init(props *properties, env environmentInfo) {
 			{
 				executable: "dotnet",
 				args:       []string{"--version"},
-				regex:      `(?:(?P<version>((?P<major>[0-9]+).(?P<minor>[0-9]+).(?:\d{2})(?P<patch>[0-9]{1}))))`,
+				regex:      `(?P<version>(?P<major>[0-9]+).(?P<minor>[0-9]+).(?P<patch>[0-9]+)(?:-(?P<rc>rc\.[a-zA-Z0-9.]*))?)`,
```

Comment:
could be something else than RC. Have to check for "real" semver regex(if one exists).

Reply:
agreed, could be anything

Reply:
I used the one from semver.org in the end. Looks good.

Reply:
https://regex101.com/r/q9rQn3/1